### PR TITLE
Support embedded-hal 0.x and 1.0-alpha.x

### PR DESCRIPTION
**Pull request by tomgilligan**
Rebased dbrgn 1-alpha branch

This moves support for different versions of embedded-hal to separate modules `eh0` and `eh1` as suggested at https://github.com/dbrgn/embedded-hal-mock/issues/70#issuecomment-1555003903

### CODE REVIEW

**Comment by rursprung** [comment]
see my comment on https://github.com/dbrgn/embedded-hal-mock/issues/70#issuecomment-1616625227 for why i (now) think that _not_ merging e-h 0 and e-h 1 mocks in the same e-h-m release would be better.

**Comment by rursprung** [comment]
fair enough. though then it might be worth it to move eh0 stuff into the eh0 module and either have all eh1 things top-level or at least re-export them at the top-level so that they don't need a breaking change in the future (and this way e-h-m can match e-h in terms of API structure). WDYT?

**Review by newAM — APPROVED** [review]
This looks great!

File: src/eh1/serial.rs
Comment (ryankurte) [inline src/eh1/serial.rs:16]: this might be how we've always done it but, it seems like it's worth having descriptive names at the module level to avoid every consumer having to `Mock as SerialMock` ?

Comment (dbrgn) [reply src/eh1/serial.rs:16]: I'd rather keep PRs focussed on one thing, makes it easier to review 🙂

The original idea was to implement the same pattern for every mock. However, I guess you're right and if every user aliases the import, we can rename the mocks as well. So: A separate PR would be welcome (but it might make sense to base it on this branch, or to wait until it is merged.)

**Comment by dbrgn** [comment]
Thanks for your work, @tomgilligan and others! This looks great at first glance! I'd like to take a closer look before merging (since it's a major change), but don't know yet when I find time to do that. Hopefully by next week... If I forget about it, feel free to ping me in this thread at any time! 🙂 (I do get notifications, but sometimes they drown in the flood of e-mails.)

**Comment by dbrgn** [comment]
I did take a closer look now, and also ported one of my project's tests to your branch. It is looking good!

Let's merge this branch, and I'll try to publish a release (maybe an `rc1`) soon. Thanks for your work!

**Review by dbrgn — APPROVED** [review]

**Comment by dbrgn** [comment]
This is now released as `embedded-hal-mock` version `0.10.0-rc.1`: https://crates.io/crates/embedded-hal-mock/0.10.0-rc.1

**Pull request merged**
